File: API-pokedex-pokemon-card/Models/Pokedex.cs
using API_pokedex_pokemon_card.Models;

public class Pokedex
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public required string ShareCode { get; set; }
    public required PokedexType Type { get; set; }
    public List<PokedexUser> PokedexUsers { get; set; } = new();
    public List<PokedexOwnedPokemonCard> OwnedPokemonCards { get; set; } = new();
    public List<PokedexWantedPokemonCard> WantedPokemonCards { get; set; } = new();
    public List<PokedexSpecificPokemon> SpecificPokemons { get; set; } = new();
}

public class PokedexCompletion
{
    public int PokedexId { get; set; }
    public int MaxPokemon { get; set; }
    public int OwnedPokemonNb { get; set; }
}


public readonly record struct Pair(int PokemonId, string PokemonCardId);

public class PokedexCompletionData
{
    public PokedexType Type { get; set; }
    public IReadOnlyList<Pair> WantedPairs { get; set; } = [];
    public IReadOnlyList<Pair> OwnedPairs { get; set; } = [];
}

public enum PokedexType
{
    LivingDex,
    SpecificPokemonsDex,
    ZarbiDex
}

File: front-pokedex-pokemon-card/src/app/Models/pokedex.ts
export interface Pokedex {
    id: number;
    userId: number;
    name: string;
    shareCode: string;
    type: PokedexType;
    specificPokemons: PokedexSpecificPokemons[];
}

export interface PokedexCreate {
    name: string;
    userId: number;
    type: PokedexType;
}

export interface PokedexUser {
    pokedexId: number;
    userId: number;
    isOwner: boolean;
    pokedex: Pokedex;
    type: PokedexType;
}

export interface PokedexCompletion {
    pokedexId: number
    maxPokemon: number
    ownedPokemonNb: number
}

export interface PokedexStats {
    title: string;
    pokedexValuationHistory: PokedexValuationHistory[];
    acquiredPriceTotal: number;
}

export interface PokedexValuationHistory {
    id: number;
    pokedexId: number;
    recordedAt: Date;
    totalValue: number;
}

export interface PokedexSpecificPokemons {
    id: number;
    pokemonId: number;
    slot: number;
}

export enum PokedexType {
    LivingDex,
    SpecificPokemonsDex 
}